### DISCUSSION TITLE
How to Implement custom proxy logic

### DISCUSSION THREAD

**Discussion by SpringHgui** [dotnet/yarp#1788] [General] [answered]
now  i am  inherited  `ForwarderHttpClientFactory` And override `ConfigureHandler` to Implement custom proxy logic
```c#
      protected override void ConfigureHandler(ForwarderHttpClientContext context, SocketsHttpHandler handler)
      {
          base.ConfigureHandler(context, handler);
          handler.ConnectCallback = ConnectCallback;
      }
```
There are other ways to do this ？

**Comment by Tratcher** [comment]
What specifically are you trying to customize?

**Comment by SpringHgui** [comment]
use repo：https://github.com/FastTunnel/FastTunnel/blob/v2/FastTunnel.Core/Forwarder/FastTunnelForwarderHttpClientFactory.cs
i override `ConfigureHandler` to customize ConnectCallback，it worked as i think very well。

but socket port can not dispose，mybe during `HttpForwarder` request has cloed，but `FastTunnelForwarderHttpClientFactory` can not perceive，how to perceive Client closed browser，so i can dispose my socket ？

**Comment by SpringHgui** [comment]
@Tratcher Can you understand what that means? what should i do to resolve this problem

**Comment by karelz** [comment]
Triage: Looks like the question is "When to close the socket when using `ConnectCallback`" - @Tratcher will respond

**Comment by Tratcher** [comment, marked as answer]
Keep in mind that the ConnectCallback is not called per request. Streams returned from the ConnectCallback go into SocketHttpHandler's connection pool and are re-used for all requests being sent to the same host. SocketHttpHandler will close the Stream when it's done with the connection.

HttpContext.RequestAborted should not be used in this callback since that is a per request event.

Even if the client closes the browser/connection, that doesn't mean the ConnectCallback stream should be closed, since it can be re-used by other clients that are sending requests to the same destination.
